packages/proxy/lib/http/util/rewriter.ts

```diff
@@ -22,10 +22,7 @@ export type InjectionOpts = {
   shouldInjectDocumentDomain: boolean
 }
 
-const doctypeRe = /<\!doctype.*?>/i
-const headRe = /<head(?!er).*?>/i
-const bodyRe = /<body.*?>/i
-const htmlRe = /<html.*?>/i
+const doctypeRe = /^\s*(?:<!--[\s\S]*?-->\s*)*(?:<\?xml[\s\S]*?\?>\s*)?(?:<!--[\s\S]*?-->\s*)*<!doctype[\s\S]*?>/i
```

Comment:
I feel like this regex is extremely complex to digest and may be even harder to fix if there are issues with it in the future. I prefer the previous approach, where we are able to clearly understand what is being matched/replaced and take the performance hit for reduced complexity